gitpod-io/gitpod
# [code] sync profiles/collections

## Description

Adds sync profile support to code-sync server

## How to test

1. Open a workspace using latest vscode in preview env https://jp-sync-collections.preview.gitpod-dev.com/workspaces
2. Update settings `"workbench.experimental.settingsProfiles.enabled": true`
3. Play around with [profiles](https://code.visualstudio.com/updates/v1_72#_settings-profiles), create, update, delete profiles and resources (settings, task, snippets) etc
4. open developer tools and check code-sync endpoint request return sucessfully
5. You can play in insider.vscode.dev and compare request responses
6. Check database migration works just fine
    - Use any other prev env and play with settings sync to add some data to the database
    - Connect to database and run DB migration manually, check it runs successfully

## Release Notes

```release-note
NONE
```

## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
